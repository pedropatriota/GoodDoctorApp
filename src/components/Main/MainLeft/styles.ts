import styled from "styled-components"
import { CheckCircle } from "@styled-icons/boxicons-regular/CheckCircle"

export const Container = styled.div`
  width: 50%;
  height: 100vh;
  background-color: #005073;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  gap: 1rem;
  padding: 3rem;
`
export const Logo = styled.h1`
  color: #71c7ec;
  text-align: center;
  font-size: 2.5rem;
`
export const Wrapper = styled.div`
  color: #fff;
  text-align: center;
  h3 {
    font-size: 2rem;
    text-align: center;
    margin-bottom: 3rem;
  }
  li {
    font-size: 1.5rem;
    text-align: left;
    margin-top: 1rem;
  }
`

export const WrapperImages = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 2rem 10rem;
`

export const ImageDoctorWoman = styled.img`
  width: min(10rem, 100%);
`
export const ImageDoctorMan = styled.img`
  width: min(18rem, 100%);
`

export const Check = styled(CheckCircle)`
  width: 2rem;
  margin-right: 1rem;
  color: #71c7ec;
`
