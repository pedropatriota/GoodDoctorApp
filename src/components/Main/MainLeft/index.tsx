import * as S from "./styles"

const MainLeft = () => (
  <S.Container>
    <S.Logo> GD - Good Doctor</S.Logo>

    <S.Wrapper>
      <h3>Sistema completo para sua pré e pós consulta:</h3>
      <ul>
        <li>
          <S.Check />
          Agendamento de consultas
        </li>
        <li>
          <S.Check />
          Receita médica, atestados e etc, todos digitais
        </li>
        <li>
          <S.Check />
          Contato direto com o médico ate 3 dias após a consulta
        </li>
        <li>
          <S.Check />
          Avaliie sua jornada, até finalizar a consulta
        </li>
      </ul>
    </S.Wrapper>

    <S.WrapperImages>
      <S.ImageDoctorWoman
        src="/img/doctor-woman.svg"
        alt="Imagem de uma médica com mascara e prancheta na mão"
      />
      <S.ImageDoctorMan
        src="/img/doctor-man.svg"
        alt="Imagem de um médico com mascara e prancheta na mão"
      />
    </S.WrapperImages>
  </S.Container>
)

export default MainLeft
