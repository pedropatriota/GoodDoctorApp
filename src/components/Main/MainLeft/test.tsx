import { render, screen } from "@testing-library/react"
import MainLeft from "./index"

describe("testing component MainLeft", () => {
  it("Should render the heading", () => {
    const { container } = render(<MainLeft />)

    expect(
      screen.getByRole("heading", {
        name: /Sistema completo para sua pré e pós consulta/i,
      })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it("should render corretly the color", () => {
    const { container } = render(<MainLeft />)

    expect(container.firstChild).toHaveStyle({ "background-color": "#005073" })
  })
})
