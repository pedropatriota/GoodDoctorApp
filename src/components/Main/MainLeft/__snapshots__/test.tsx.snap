// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing component MainLeft Should render the heading 1`] = `
<div
  class="styles__Container-hXcVkP hIkssT"
>
  <img
    alt="logo"
    class="styles__Logo-bRSJPP cBIDup"
    src="/img/logo.svg"
  />
  <div
    class="styles__Wrapper-gFJSv dAVBHL"
  >
    <h3>
      Sistema completo para sua pré e pós consulta:
    </h3>
    <ul>
      <li>
        <svg
          aria-hidden="true"
          class="StyledIconBase-ea9ulj-0 cKCFNq styles__Check-fLwBCG hQTzIs"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"
          />
          <path
            d="M9.999 13.587L7.7 11.292l-1.412 1.416 3.713 3.705 6.706-6.706-1.414-1.414z"
          />
        </svg>
        Agendamento de consulta
      </li>
      <li>
        <svg
          aria-hidden="true"
          class="StyledIconBase-ea9ulj-0 cKCFNq styles__Check-fLwBCG hQTzIs"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"
          />
          <path
            d="M9.999 13.587L7.7 11.292l-1.412 1.416 3.713 3.705 6.706-6.706-1.414-1.414z"
          />
        </svg>
        Recita médica, atestados e etc, todos digitais
      </li>
      <li>
        <svg
          aria-hidden="true"
          class="StyledIconBase-ea9ulj-0 cKCFNq styles__Check-fLwBCG hQTzIs"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"
          />
          <path
            d="M9.999 13.587L7.7 11.292l-1.412 1.416 3.713 3.705 6.706-6.706-1.414-1.414z"
          />
        </svg>
        Contato direto com o médico ate 3 dias após a consulta
      </li>
      <li>
        <svg
          aria-hidden="true"
          class="StyledIconBase-ea9ulj-0 cKCFNq styles__Check-fLwBCG hQTzIs"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"
          />
          <path
            d="M9.999 13.587L7.7 11.292l-1.412 1.416 3.713 3.705 6.706-6.706-1.414-1.414z"
          />
        </svg>
        Avaliação do paciente através de um simples formulário
      </li>
    </ul>
  </div>
  <div
    class="styles__WrapperImages-hWToWb hLCeSh"
  >
    <img
      alt="Imagem de uma médica com mascara e prancheta na mão"
      class="styles__ImageDoctorWoman-kCDKzg zCFIl"
      src="/img/doctor-woman.svg"
    />
    <img
      alt="Imagem de um médico com mascara e prancheta na mão"
      class="styles__ImageDoctorMan-dCBvwA ibcLwK"
      src="/img/doctor-man.svg"
    />
  </div>
</div>
`;
